pkg/instrumentation/upgrade/upgrade.go

```diff
@@ -74,97 +88,54 @@ func (u *InstrumentationUpgrade) ManagedInstances(ctx context.Context) error {
-			}
-		} else {
-			u.Logger.Error(nil, "support for Python auto instrumentation is not enabled")
-			u.Recorder.Event(inst.DeepCopy(), "Warning", "InstrumentationUpgradeRejected", "support for Python auto instrumentation is not enabled")
-		}
-	}
-	autoInstDotnet := inst.Annotations[v1alpha1.AnnotationDefaultAutoInstrumentationDotNet]
-	if autoInstDotnet != "" {
-		if featuregate.EnableDotnetAutoInstrumentationSupport.IsEnabled() {
-			// upgrade the image only if the image matches the annotation
-			if inst.Spec.DotNet.Image == autoInstDotnet {
-				inst.Spec.DotNet.Image = u.DefaultAutoInstDotNet
-				inst.Annotations[v1alpha1.AnnotationDefaultAutoInstrumentationDotNet] = u.DefaultAutoInstDotNet
-			}
-		} else {
-			u.Logger.Error(nil, "support for .NET auto instrumentation is not enabled")
-			u.Recorder.Event(inst.DeepCopy(), "Warning", "InstrumentationUpgradeRejected", "support for .NET auto instrumentation is not enabled")
-		}
-	}
-	autoInstGo := inst.Annotations[v1alpha1.AnnotationDefaultAutoInstrumentationGo]
-	if autoInstGo != "" {
-		if featuregate.EnableGoAutoInstrumentationSupport.IsEnabled() {
-			// upgrade the image only if the image matches the annotation
-			if inst.Spec.Go.Image == autoInstGo {
-				inst.Spec.Go.Image = u.DefaultAutoInstGo
-				inst.Annotations[v1alpha1.AnnotationDefaultAutoInstrumentationGo] = u.DefaultAutoInstGo
-			}
-		} else {
-			u.Logger.Error(nil, "support for Go auto instrumentation is not enabled")
-			u.Recorder.Event(inst.DeepCopy(), "Warning", "InstrumentationUpgradeRejected", "support for Go auto instrumentation is not enabled")
-		}
-	}
-	autoInstApacheHttpd := inst.Annotations[v1alpha1.AnnotationDefaultAutoInstrumentationApacheHttpd]
-	if autoInstApacheHttpd != "" {
-		if featuregate.EnableApacheHTTPAutoInstrumentationSupport.IsEnabled() {
-			// upgrade the image only if the image matches the annotation
-			if inst.Spec.ApacheHttpd.Image == autoInstApacheHttpd {
-				inst.Spec.ApacheHttpd.Image = u.DefaultAutoInstApacheHttpd
-				inst.Annotations[v1alpha1.AnnotationDefaultAutoInstrumentationApacheHttpd] = u.DefaultAutoInstApacheHttpd
-			}
-		} else {
-			u.Logger.Error(nil, "support for Apache HTTPD auto instrumentation is not enabled")
-			u.Recorder.Event(inst.DeepCopy(), "Warning", "InstrumentationUpgradeRejected", "support for Apache HTTPD auto instrumentation is not enabled")
-		}
-	}
-	autoInstNginx := inst.Annotations[v1alpha1.AnnotationDefaultAutoInstrumentationNginx]
-	if autoInstNginx != "" {
-		if featuregate.EnableNginxAutoInstrumentationSupport.IsEnabled() {
-			// upgrade the image only if the image matches the annotation
-			if inst.Spec.Nginx.Image == autoInstNginx {
-				inst.Spec.Nginx.Image = u.DefaultAutoInstNginx
-				inst.Annotations[v1alpha1.AnnotationDefaultAutoInstrumentationNginx] = u.DefaultAutoInstNginx
+func (u *InstrumentationUpgrade) upgrade(_ context.Context, inst v1alpha1.Instrumentation) *v1alpha1.Instrumentation {
```

Comment:
the previous way was really annoying me. If anyone has problems with this, i am happy to revert it.